infra/metadata.yaml

```diff
@@ -19,99 +19,113 @@ metadata:
+    - name: simple_example
+      location: examples/simple_example
+  interfaces:
+    variables:
+    - name: client_image_host
+      description: Container Registry that hosts the client image (PROJECT_ID[/folder])
+      varType: string
+      defaultValue: hsa-public/terraform-python-dynamic-webapp
+    - name: database_name
+      description: Cloud SQL database name
+      varType: string
+      defaultValue: django
+    - name: database_username
+      description: Cloud SQL database name
+      varType: string
+      defaultValue: server
+    - name: image_version
+      description: Version of the Container Registry image to use
+      varType: string
+      defaultValue: v1.7.0
+    - name: init
+      description: Initialize database?
+      varType: bool
+      defaultValue: true
+    - name: instance_name
+      description: Cloud SQL Instance name
+      varType: string
+      defaultValue: psql
+    - name: labels
+      description: A set of key/value label pairs to assign to the resources deployed by this blueprint.
+      varType: map(string)
+      defaultValue: {}
+    - name: project_id
+      description: Google Cloud Project ID
+      varType: string
+      required: true
+    - name: random_suffix
+      description: Add random suffix to VM name
+      varType: string
+      defaultValue: true
+    - name: region
+      description: Google Cloud Region
+      varType: string
+      defaultValue: us-central1
+    - name: server_image_host
+      description: Container Registry that hosts the server image (PROJECT_ID[/folder])
+      varType: string
+      defaultValue: hsa-public/terraform-python-dynamic-webapp
+    - name: service_name
+      description: Cloud Run service name
+      varType: string
+      defaultValue: server
+    - name: zone
+      description: GCP zone for provisioning zonal resources.
+      varType: string
+      defaultValue: us-central1-c
+    outputs:
+    - name: django_admin_password
+      description: Djando Admin password
+    - name: django_admin_url
+      description: Djando Admin URL
+    - name: firebase_url
+      description: Firebase URL
+    - name: neos_toc_url
+      description: Neos Tutorial URL
+    - name: usage
+      description: Next steps for usage
+  requirements:
     roles:
-    - roles/owner
-  services:
-  - cloudresourcemanager.googleapis.com
-  - storage-api.googleapis.com
-  - serviceusage.googleapis.com
+    - level: Project
+      roles:
+      - roles/cloudsql.admin
+      - roles/compute.admin
+      - roles/compute.networkAdmin
+      - roles/firebase.managementServiceAgent
+      - roles/firebasehosting.admin
+      - roles/iam.serviceAccountAdmin
+      - roles/iam.serviceAccountUser
+      - roles/resourcemanager.projectIamAdmin
+      - roles/run.admin
+      - roles/secretmanager.admin
+      - roles/storage.admin
+    services:
+    - cloudresourcemanager.googleapis.com
+    - storage-api.googleapis.com
+    - serviceusage.googleapis.com
```

Comment:
question: I see only three services listed here, shouldn't this include all the services needed?

Reply:
I believe these are the services required to invoke terraform, which will then itself enable the other services. They're similar to the services [avocano gets you to manually enable before running the script](https://github.com/GoogleCloudPlatform/avocano/blob/main/README.walkthrough.md?plain=1#L28)

Reply:
The source for this is the [`activate_apis`](https://github.com/GoogleCloudPlatform/terraform-dynamic-python-webapp/blob/main/infra/test/setup/main.tf#L27) variable to the project factory module under test/.